samples/cpp/stitching_detailed.cpp

```diff
@@ -824,8 +828,13 @@ int main(int argc, char* argv[])
         if (timelapse)
         {
             timelapser->process(img_warped_s, Mat::ones(img_warped_s.size(), CV_8UC1), corners[img_idx]);
-
-            imwrite("fixed_" + img_names[img_idx], timelapser->getDst());
+            String fileName(img_names[img_idx]);
+            size_t posExt=fileName.find_last_of('.');
+            if (posExt!=string::npos)
+            {
+                fileName = fileName.substr(0,posExt)+"_fixed"+fileName.substr(posExt,fileName.length()-posExt);
+                imwrite(fileName, timelapser->getDst());
+            }
```

Comment:
Agreed for cases like "C:\path\image_000x.jpg", but it is better to preserve existed behavior.
Something like this (code is not checked actually):
```
size_t posFileName = std::max(fileName.find_last_of('/') + 1, fileName.find_last_of('\\') + 1); // don't optimize +1
fileName = fileName.substr(0, posFileName) + "fixed_" + fileName.substr(posFileName, fileName.length() - posFileName);
imwrite(fileName, timelapser->getDst());
```